File: HandBallStats_api/src/main/java/es/diego/handballstats_api/models/Equipo.java
package es.diego.handballstats_api.models;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

import com.fasterxml.jackson.annotation.JsonBackReference;
import com.fasterxml.jackson.annotation.JsonManagedReference;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name ="equipo", uniqueConstraints = {@UniqueConstraint(columnNames = {"foto"})})
public class Equipo {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;

    @Column(name = "nombre", nullable = false)
    private String nombre;

    @Column(name = "foto")
    private String foto;

    @Column(name = "categoria")
    private String categoria;

    @ManyToOne
    @JoinColumn(name = "id_entrenador", nullable = false)
    @JsonBackReference("a")
    private Entrenador entrenador;

    @OneToMany(mappedBy = "equipo", cascade = CascadeType.ALL)
    @JsonManagedReference("b")
    private List<Jugador> jugadores;

    @OneToMany(mappedBy = "equipo", cascade = CascadeType.ALL, orphanRemoval = true)
    @JsonManagedReference("c")
    private List<Partido> partidos;
}

File: HandBallStats/app/src/main/java/es/diego/handballstats/models/Equipo.kt
package es.diego.handballstats.models

import com.google.gson.annotations.Expose

data class Equipo(
    val id: Int?,
    var nombre: String,
    var categoria: String,
    var foto: String?,
    var entrenador: Entrenador?,
    var jugadores: MutableList<Jugador>? = mutableListOf(),
    var partidos: MutableList<Partido>? = mutableListOf()
) {
    override fun toString(): String {
        return nombre + ' ' + categoria + " entrenador: ${entrenador?.nombre}"
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other !is Equipo) return false
        return this.id == other.id
    }

    override fun hashCode(): Int {
        return id ?: 0
    }
}
